kcmdline.c

```diff
@@ -65,7 +65,7 @@ EFI_STATUS check_cmdline(CONST CHAR8 *cmdline, UINTN cmdline_len) {
 	for (i = 0; i < cmdline_len; i++) {
 		c = buf[i];
 		if (c < 0x20 || c > 0x7e) {
-			Print(L"Bad character 0x%02hhx.\n", buf);
+			Print(L"Bad character 0x%02hhx in position %d: %a.\n", c, i, cmdline);
```

Comment:
Is 0x%02hhx the right format specifier for 'c'?

Reply:
If you have a better way, i'm happy to take it. easy enough to test now.
Do you want it prefixed with '0x' to be more clear?

```c
#include <stdio.h>
#include <string.h>

int main(int argc, char* argv[]) {
    size_t s = strlen(argv[1]);
    char c;

    for (int i=0; i < argc; i++) {
        s = strlen(argv[i]);
        for (int j=0; j < s; j++) {
            c = argv[i][j];
            printf("%02hhx %c|\n", c, c);
        }
        printf("\n");
    }
    return 0;
}
```

```
$ gcc -o testme testme.c
$ ./testme "hey does that work?"
2e .|
2f /|
74 t|
65 e|
73 s|
74 t|
6d m|
65 e|

68 h|
65 e|
79 y|
20  |
64 d|
6f o|
65 e|
73 s|
20  |
74 t|
68 h|
61 a|
74 t|
20  |
77 w|
6f o|
72 r|
6b k|
3f ?|
```

Reply:
never mind. the one in the code is prefixed with 0x.
here is output of test:

```
09 | PASS | reject-S | reject-S |init=/bin/bash|
Bad character 0x09 in position 5: root= atomix.
```

Reply:
and here it is from EFI (there is a tab there).
```
Shell> fs0:
FS0:\> cd fs0:\efi\boot
FS0:\efi\boot\> launch.nsh
FS0:\efi\boot\> kernel.efi root=atomix  console=ttyS0
Bad character 0x09 in position 12: root=atomix  console=ttyS0.
```